go/keymanager/api/secret.go

```diff
@@ -42,38 +106,47 @@ func (s *EncryptedEphemeralSecret) SanityCheck(epoch beacon.EpochTime, reks map[
 		return fmt.Errorf("keymanager: sanity check failed: ephemeral secret contains an invalid epoch: (expected: %d, got: %d)", epoch, s.Epoch)
 	}
 
-	// Secret should be encrypted to at least one member of the key manager committee.
-	if len(reks) == 0 {
-		return fmt.Errorf("keymanager: sanity check failed: ephemeral secret cannot be encrypted to an empty committee")
+	if err := s.Secret.SanityCheck(reks); err != nil {
+		return err
 	}
 
-	// Secret should be encrypted to the enclaves from the key manager committee only.
-	for rek := range s.Ciphertexts {
-		if _, ok := reks[rek]; !ok {
-			return fmt.Errorf("keymanager: sanity check failed: ephemeral secret contains unknown REK")
-		}
+	return nil
+}
+
+// SignedEncryptedMasterSecret is a RAK signed encrypted master secret.
+type SignedEncryptedMasterSecret struct {
+	// Secret is the encrypted master secret.
+	Secret EncryptedMasterSecret `json:"secret"`
+
+	// Signature is a signature of the master secret.
+	Signature signature.RawSignature `json:"signature"`
+}
+
+// Verify sanity checks the encrypted master secret and verifies its signature.
+func (s *SignedEncryptedMasterSecret) Verify(generation uint64, epoch beacon.EpochTime, reks map[x25519.PublicKey]struct{}, rak *signature.PublicKey) error {
+	if err := s.Secret.SanityCheck(generation, epoch, reks); err != nil {
+		return err
 	}
 
-	// Most of the enclaves should be able to decrypt the secret.
-	percent := len(s.Ciphertexts) * 100 / len(reks)
-	if percent < minEnclavesPercent {
-		return fmt.Errorf("keymanager: sanity check failed: ephemeral secret is not encrypted to enough enclaves")
+	raw := cbor.Marshal(s.Secret)
+	if !rak.Verify(EncryptedMasterSecretSignatureContext, raw, s.Signature[:]) {
+		return fmt.Errorf("keymanager: sanity check failed: master secret contains an invalid signature")
 	}
 
 	return nil
 }
 
 // SignedEncryptedEphemeralSecret is a RAK signed encrypted ephemeral secret.
 type SignedEncryptedEphemeralSecret struct {
-	// Secret is the encrypted ephemeral secret to which the signature belongs.
+	// Secret is the encrypted ephemeral secret.
 	Secret EncryptedEphemeralSecret `json:"secret"`
 
 	// Signature is a signature of the ephemeral secret.
 	Signature signature.RawSignature `json:"signature"`
 }
 
 // Verify sanity checks the encrypted ephemeral secret and verifies its signature.
-func (s *SignedEncryptedEphemeralSecret) Verify(epoch beacon.EpochTime, reks map[x25519.PublicKey]struct{}, rak signature.PublicKey) error {
+func (s *SignedEncryptedEphemeralSecret) Verify(epoch beacon.EpochTime, reks map[x25519.PublicKey]struct{}, rak *signature.PublicKey) error {
```

Comment:
Why this change?

Reply:
I plan to introduce a new (helper) function which fetches `rak` key from node's capabilities. The key will be passed to the `Verifiy` function meaning that the array will be copied at least twice. I thought that was too much and decided to use pointers instead, especially as `rak`s are read-only. Should I revert changes?